src/main/kotlin/no/nav/familie/ba/sak/kjerne/fødselshendelse/vilkårsvurdering/Regler.kt

```diff
@@ -3,48 +3,58 @@ package no.nav.familie.ba.sak.kjerne.fødselshendelse.vilkårsvurdering
 import no.nav.familie.ba.sak.kjerne.grunnlag.personopplysninger.statsborgerskap.GrStatsborgerskap
 import no.nav.familie.kontrakter.felles.personopplysning.OPPHOLDSTILLATELSE
 import no.nav.familie.kontrakter.felles.personopplysning.SIVILSTAND
 import java.time.Duration
 import java.time.LocalDate
 
-fun vurderPersonErBosattIRiket(adresse: GrBostedsadresse?): Evaluering {
+fun vurderPersonErBosattIRiket(adresser: MutableList<GrBostedsadresse>, vurderFra: LocalDate): Evaluering {
     /**
      * En person med registrert bostedsadresse er bosatt i Norge.
      * En person som mangler registrert bostedsadresse er utflyttet.
      * See: https://navikt.github.io/pdl/#_utflytting
      */
-    return adresse?.let { Evaluering.oppfylt(VilkårOppfyltÅrsak.BOR_I_RIKET) }
-           ?: Evaluering.ikkeOppfylt(VilkårIkkeOppfyltÅrsak.BOR_IKKE_I_RIKET)
+    return if (adresser.isNotEmpty() && hentMaxAvstandAvDagerMellomPerioder(adresser.mapNotNull { it.periode },
+                                                                            vurderFra,
+                                                                            LocalDate.now()) == 0L)
+        Evaluering.oppfylt(VilkårOppfyltÅrsak.BOR_I_RIKET)
+    else Evaluering.ikkeOppfylt(VilkårIkkeOppfyltÅrsak.BOR_IKKE_I_RIKET)
 }
 
 fun vurderPersonErUnder18(alder: Int): Evaluering =
         if (alder < 18) Evaluering.oppfylt(VilkårOppfyltÅrsak.ER_UNDER_18_ÅR)
         else Evaluering.ikkeOppfylt(VilkårIkkeOppfyltÅrsak.ER_IKKE_UNDER_18_ÅR)
 
 
-fun vurderBarnetErBosattMedSøker(søkerAdresse: GrBostedsadresse?, barnAdresse: GrBostedsadresse?): Evaluering =
-        when {
-            GrBostedsadresse.erSammeAdresse(søkerAdresse, barnAdresse) -> Evaluering.oppfylt(
-                    VilkårOppfyltÅrsak.BARNET_BOR_MED_MOR)
-            else -> Evaluering.ikkeOppfylt(VilkårIkkeOppfyltÅrsak.BARNET_BOR_IKKE_MED_MOR)
-        }
+fun vurderBarnetErBosattMedSøker(søkerAdresser: MutableList<GrBostedsadresse>,
+                                 barnAdresser: MutableList<GrBostedsadresse>): Evaluering {
+
+    return if (barnAdresser.isNotEmpty() && barnAdresser.all {
+                val søkerAdressePåSammeTidspunkt =
+                        søkerAdresser.firstOrNull { søkerAdresse -> søkerAdresse.periode?.fom == it.periode?.fom && søkerAdresse.periode?.tom == it.periode?.tom }
+
+                if (søkerAdressePåSammeTidspunkt == null) false
+                else GrBostedsadresse.erSammeAdresse(søkerAdressePåSammeTidspunkt, it)
+            }) Evaluering.oppfylt(
+            VilkårOppfyltÅrsak.BARNET_BOR_MED_MOR)
+    else Evaluering.ikkeOppfylt(VilkårIkkeOppfyltÅrsak.BARNET_BOR_IKKE_MED_MOR)
+}
 
-fun vurderPersonErUgift(sivilstand: GrSivilstand?): Evaluering {
-    return when (sivilstand?.type) {
-        SIVILSTAND.UOPPGITT ->
+fun vurderPersonErUgift(sivilstand: List<GrSivilstand>): Evaluering {
+    return when {
+        sivilstand.any { it.type == SIVILSTAND.UOPPGITT } ->
```

Comment:
```suggestion
        sivilstand.singleOrNull { it.type == SIVILSTAND.UOPPGITT } != null ->
```
skal vel egentlig være unødvendig siden datakvaliteten skal vel være god, men skader ikke å barnesikre siden det skal kjøre automatisk

Reply:
Det samme gjelder forsåvidt på anysjekken her https://github.com/navikt/familie-ba-sak/pull/1302/files#diff-b3eaac9f549cba002b0524186af567f4c8640d1182c9310b3d7302c482e7e947R57. 

Kanskje man heller skulle lagt til noe sånt som første when: 
sivilstand.size > 1 -> throw Feil("Barn har flere sivilstander registrert ved fødsel")

Reply:
Bra, dette er gull 👍

Reply:
Ettersom vi potensielt kan gjenbruke dette i manuell behandling tenker jeg vi prøver å etterligne vilkårene best mulig.